File: janari0_mobile/lib/model/product.dart
import 'package:janari0/model/photo.dart';
import 'package:json_annotation/json_annotation.dart';

part 'product.g.dart';

@JsonSerializable()
class Product {
  int? productId;
  String? name;
  DateTime? expirationDate;
  late List<Photo> photos;

  Product(
      {required this.productId,
      this.name,
      this.expirationDate,
      required this.photos});

  factory Product.fromJson(Map<String, dynamic> json) =>
      _$ProductFromJson(json);

  Map<String, dynamic> toJson() => _$ProductToJson(this);

  static List<Product> fromJsonList(List list) {
    return list.map((item) => Product.fromJson(item)).toList();
  }

  ///this method will prevent the override of toString
  String userAsString() {
    return '#$productId $name';
  }

  ///custom comparing function to check if two users are equal
  bool isEqual(Product model) {
    return productId == model.productId;
  }

  @override
  String toString() => name!;
}

File: Janari0.Model/Product.cs
using System;
using System.Collections.Generic;

namespace Janari0.Model
{
    public partial class Product
    {
        public int ProductId { get; set; }
        public string Name { get; set; }
        public DateTime ExpirationDate { get; set; }
        public int UserId { get; set; }
        public virtual List<Photo> Photos { get; set; } = new List<Photo>();
        public virtual ProductsSale ProductsSale { get; set; }
        public virtual User User { get; set; }
    }
}
